src/chain.cpp

```diff
@@ -61,6 +61,20 @@ const CBlockIndex *CChain::FindFork(const CBlockIndex *pindex) const {
     return pindex;
 }
 
+
+struct SmallerThanTimestamp {
+    bool operator()(CBlockIndex* pBlock, const int64_t& time) const
+    {
+        return pBlock->GetBlockTime() < time;
+    }
+};
+
+CBlockIndex* CChain::FindLatestBefore(int64_t nTime) const
+{
+    std::vector<CBlockIndex*>::const_iterator lower = std::lower_bound(vChain.begin(), vChain.end(), nTime, SmallerThanTimestamp());
```

Comment:
Using c++11 lambda here?

Something like:
```c++
std::lower_bound(vChain.begin(), vChain.end(), nTime,
   [](CBlockIndex* pBlock, const int64_t& time) -> bool { return pBlock->GetBlockTime() < time; }
);
```

Would probably be less confusing to read.